### DISCUSSION TITLE
NETBOX SECRET STORE plugin error 'str' object has no attribute '_meta'

### DISCUSSION THREAD

**Discussion by vinilex77** [netbox-community/netbox#9995] [Q&A]
Hi everyone, after installed NETBOX SECRET STORE plugin i was made to add a new user key.
Insert my public RSA key and when click to Create button this message comes out:

![immagine](https://user-images.githubusercontent.com/110663678/184362062-084a11d0-dd03-4e6b-94ba-74c7ce7b5532.png)

'str' object has no attribute '_meta'

the only part that works is secret roles, add correctly.

below the logs:
Aug 11 07:37:14 netbox gunicorn[907]: Internal Server Error: /plugins/netbox_secretstore/user-key/edit/
Aug 11 07:37:14 netbox gunicorn[907]: Traceback (most recent call last):
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
Aug 11 07:37:14 netbox gunicorn[907]: response = get_response(request)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
Aug 11 07:37:14 netbox gunicorn[907]: response = wrapped_callback(request, *callback_args, **callback_kwargs)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 84, in view
Aug 11 07:37:14 netbox gunicorn[907]: return self.dispatch(request, *args, **kwargs)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/netbox_secretstore/views.py", line 269, in dispatch
Aug 11 07:37:14 netbox gunicorn[907]: return super().dispatch(request, *args, **kwargs)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
Aug 11 07:37:14 netbox gunicorn[907]: return super().dispatch(request, *args, **kwargs)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 119, in dispatch
Aug 11 07:37:14 netbox gunicorn[907]: return handler(request, *args, **kwargs)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/netbox_secretstore/views.py", line 289, in post
Aug 11 07:37:14 netbox gunicorn[907]: return render(request, self.template_name, {
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/shortcuts.py", line 24, in render
Aug 11 07:37:14 netbox gunicorn[907]: content = loader.render_to_string(template_name, context, request, using=using)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/loader.py", line 62, in render_to_string
Aug 11 07:37:14 netbox gunicorn[907]: return template.render(context, request)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/backends/django.py", line 62, in render
Aug 11 07:37:14 netbox gunicorn[907]: return self.template.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 175, in render
Aug 11 07:37:14 netbox gunicorn[907]: return self._render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
Aug 11 07:37:14 netbox gunicorn[907]: return self.nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]: return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render
Aug 11 07:37:14 netbox gunicorn[907]: return compiled_parent._render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
Aug 11 07:37:14 netbox gunicorn[907]: return self.nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]: return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render
Aug 11 07:37:14 netbox gunicorn[907]: return compiled_parent._render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
Aug 11 07:37:14 netbox gunicorn[907]: return self.nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]: return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render
Aug 11 07:37:14 netbox gunicorn[907]: return compiled_parent._render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
Aug 11 07:37:14 netbox gunicorn[907]: return self.nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]: return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 63, in render
Aug 11 07:37:14 netbox gunicorn[907]: result = block.nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]: return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/defaulttags.py", line 322, in render
Aug 11 07:37:14 netbox gunicorn[907]: return nodelist.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in
Aug 11 07:37:14 netbox gunicorn[907]: return SafeString("".join([node.render_annotated(context) for node in self]))
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
Aug 11 07:37:14 netbox gunicorn[907]: return self.render(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 1059, in render
Aug 11 07:37:14 netbox gunicorn[907]: output = self.filter_expression.resolve(context)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox-3.2.7/venv/lib/python3.8/site-packages/django/template/base.py", line 739, in resolve
Aug 11 07:37:14 netbox gunicorn[907]: new_obj = func(obj, *arg_vals)
Aug 11 07:37:14 netbox gunicorn[907]: File "/opt/netbox/netbox/utilities/templatetags/builtins/filters.py", line 79, in meta
Aug 11 07:37:14 netbox gunicorn[907]: return getattr(model._meta, attr, '')
Aug 11 07:37:14 netbox gunicorn[907]: AttributeError: 'str' object has no attribute '_meta'
Aug 11 07:37:14 netbox gunicorn[907]: Not Found: /favicon.ico

Best regards!

**Comment by DanSheps** [comment]
What size of a key are you using?

**Reply by vinilex77** [reply]
Hello Dan, Thank you for the response.  Sorry I'm late, but I was on vacation.

i try several key lenght.
For example i try to generate from this:
https://travistidwell.com/jsencrypt/demo/

**Comment by Sweepers007** [comment]
Hi Vinilex,  did you solve this issue? If so how?  Im on netbox 3.6.2 and netbox-secrets 1.9.0 with the same issue.

Thanks
Jaco
